### ISSUE TITLE
Cannot communicate with DRV8303 over SPI

### ISSUE THREAD

**Issue by jonnyando** [jonnyando/AGV#1]
Still debugging, but so far cannot see any data on the SDO line of the bus.

Current transmission looks like:
1. Lower chip select
2. transfer packet (R/W=1, address = 0010, data = 0)
3. Raise CS, delay 1mS, Lower CS
4. run SPI1_Tranceive(0x0000); to send zeroes and expecting to receive data back.

When running the above, only zeroes can be seen on the SDO line. No pulsing at all. Need to check IO configuration, but I expect it is correct.
Next things to check are baudrate, phase, and polarity of the clock.

**Comment by jonnyando** [comment]
How it's initialised by ODrive firmware:
:   hspi3.Instance = SPI3;
   66:   hspi3.Init.Mode = SPI_MODE_MASTER;
   67:   hspi3.Init.Direction = SPI_DIRECTION_2LINES;
   68:   hspi3.Init.DataSize = SPI_DATASIZE_16BIT;
   69:   hspi3.Init.CLKPolarity = SPI_POLARITY_LOW;
   70:   hspi3.Init.CLKPhase = SPI_PHASE_2EDGE;
   71:   hspi3.Init.NSS = SPI_NSS_SOFT;
   72:   hspi3.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_16;
   73:   hspi3.Init.FirstBit = SPI_FIRSTBIT_MSB;
   74:   hspi3.Init.TIMode = SPI_TIMODE_DISABLE;
   75:   hspi3.Init.CRCCalculation = SPI_CRCCALCULATION_DISABLE;
   76:   hspi3.Init.CRCPolynomial = 10;

**Comment by jonnyando** [comment]
Issue has been resolved.

There are two issues at play here - both hardware and software - with hardware being the biggest cause.

In the schematic, the placement of pins 23 and 24 are reversed, meaning AVDD is shorted directly to ground. Cutting the trace between that pin and the cap/via that bypasses it fixes the problem. The fix will be made in the schematic and new boards will be ordered soon.

For the software, I still haven't been able to figure out what hardware controlled NSS means, but I think it's only for Slave mode, so I've been using software controlled. The actual problem is the TX_empty flag gets raised before the transmission has completed on the hardware, so using that as a blocker to then raise chip select doesnt work and chip select get's raised about halfway through the transmission. I've implemented a NOP loop for the right amount of cycles, but will try to implement a proper delay function.
